File: Python/Komponen.py

class Komponen:
    def __init__(self, merk="", nama=""):
        self._merk = merk
        self._nama = nama


    # getter
    def get_merk(self):
        return self._merk

    def get_nama(self):
        return self._nama

    # setter
    def set_merk(self, merk):
        self._merk = merk

    def set_nama(self, nama):
        self._nama = nama

    # method info
    def info(self):
        return f"Komponen : {self._nama} (Merk : {self._merk})"

File: CPP/Komponen.cpp
#include <bits/stdc++.h>
#include <string>
#include <vector>
#include <iostream>
#pragma once

using namespace std;


class Komponen
{
protected:
    string merk;
    string nama;

public:
    // konstruktor
    Komponen(string merk = "", string nama = "")
    {
        this->merk = merk;
        this->nama = nama;
    }
    // Destruktor
    ~Komponen()
    {
    }
    // getter
    string getMerk()
    {
        return merk;
    }
    string getNama()
    {
        return nama;
    }
    // setter
    void setMerk(string merk)
    {
        this->merk = merk;
    }
    void setNama(string nama)
    {
        this->nama = nama;
    }

    // method info
    virtual string info()
    {
        return "Komponen : " + nama + "(Merk : " + merk + ")";
    }
};